[8.19] ESQL: Disallow mixed quoted/unquoted patterns in FROM (#127636)

pawankartik-elastic opened the pull request:
This will backport the following commits from `main` to `8.19`:
 - [ESQL: Disallow mixed quoted/unquoted patterns in FROM (#127636)](https://github.com/elastic/elasticsearch/pull/127636)

<!--- Backport version: 9.6.6 -->

alex-spies committed:
ESQL: Disallow mixed quoted/unquoted patterns in FROM (#127636)

Disallow partial quoting: individual quoting of constituent strings in an index pattern

---------

Co-authored-by: Pawan Kartik <pawankartik.chitrapu@elastic.co>
(cherry picked from commit 90210406e05a6a98cedeab6376ae3f8e382e38ff)

# Conflicts:
#	x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/EsqlBaseParser.interp
#	x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/EsqlBaseParser.java
#	x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/IdentifierGenerator.java

pawankartik-elastic commented:
I've verified that the ANTLR Java classes that are auto-generated are up-to-date by explicitly switching to this branch and running `./gradlew ":x-pack:plugin:esql:regen"`.

pawankartik-elastic committed:
Merge branch '8.19' into backport/8.19/pr-127636

idegtiarenko commented:
Do you mind including the same fix in this backport: https://github.com/elastic/elasticsearch/pull/129611 ?

pawankartik-elastic committed:
Fix CI

pawankartik-elastic commented:
I've pulled the fix into this backport and fixed the CI. Turns out the `TS` command is still `METRICS` in this branch and that affected the line number and offsets in the error messages — this caused the CI failure. Also, the grammar is different from what's on `9.1` and that's causing different messages to appear. So I've adjusted the error messages.

Requested explicit approval for this backport for these reasons.
